src/parcsr_mv/gen_fffc.c

```diff
@@ -346,6 +346,12 @@ hypre_ParCSRMatrixGenerateFFFC( hypre_ParCSRMatrix  *A,
          {
             fpt = fpt_array[i];
             fpt2 = fpt_array[i-1];
+
```

Comment:
A simple fix but hard to find. In the cases that `num_threads` is large enough, a thread can have no F-pt in its "partition", so we must skip the adds in the rest of the loop.